### PR TITLE
Reuse `StationEui` and `JoinEui`

### PR DESCRIPTION
**Pull request by bastbu**
# PR for issue #964

## What is being addressed

Reuses `StationEui` and `JoinEui`.

### CODE REVIEW

File: Tests/Simulation/IntegrationTestFixtureSim.cs [hunk 1]
```diff
@@ -53,7 +53,7 @@ public override void SetupTestDevices()
             Device1002_Simulated_OTAA = new TestDeviceInfo()
             {
                 DeviceID = "0000000000001002",
-                AppEUI = "0000000000001002",
+                AppEUI = GetJoinEui(1002),
```
Comment (atifaziz) [inline Tests/Simulation/IntegrationTestFixtureSim.cs:56]: `0000000000001002` was in hex so shouldn't this read `GetJoinEui(0x1002)`?

File: Tests/Simulation/IntegrationTestFixtureSim.cs [hunk 2]
```diff
@@ -64,7 +64,7 @@ public override void SetupTestDevices()
             Device1003_Simulated_HttpBasedDecoder = new TestDeviceInfo
             {
                 DeviceID = "0000000000001003",
-                AppEUI = "0000000000001003",
+                AppEUI = GetJoinEui(1003),
```
Comment (atifaziz) [inline Tests/Simulation/IntegrationTestFixtureSim.cs:67]: `0000000000001003` was in hex so shouldn't this read `GetJoinEui(0x1003)`?

File: Tests/Simulation/IntegrationTestFixtureSim.cs (hunk 1 above)
Comment (bastbu) [reply Tests/Simulation/IntegrationTestFixtureSim.cs:56]: Yes, it is not the same value before/after. I tried to be consistent with some of the other helper methods, where in the past we went away from passing hex numbers to just passing integers in decimal representation. Based on our discussion offline, we should probably revisit how we create these devices entirely.

File: Tests/E2E/OTAAJoinTest.cs [hunk 3]
```diff
@@ -142,7 +142,7 @@ public async Task OTAA_Join_With_Wrong_AppEUI_Fails()
             var device = TestFixtureCi.Device13_OTAA;
             LogTestStart(device);
 
-            var appEUIToUse = "FF7A00000000FCE3";
+            var appEUIToUse = JoinEui.Parse("FF7A00000000FCE3");
```
Comment (atifaziz) [inline Tests/E2E/OTAAJoinTest.cs:145]: nit:

```suggestion
            var appEUIToUse = new JoinEui(0xFF7A00000000FCE3);
```

File: LoRaEngine/LoraKeysManagerFacade/ConcentratorCredentialsFunction.cs [hunk 4]
```diff
@@ -62,12 +64,13 @@ public class ConcentratorCredentialsFunction
...
         {
-            var stationEui = req.Query["StationEui"];
-            if (StringValues.IsNullOrEmpty(stationEui))
+            var rawStationEui = req.Query["StationEui"];
+            if (StringValues.IsNullOrEmpty(rawStationEui))
             {
                 this.logger.LogError("StationEui missing in request");
                 return new BadRequestObjectResult("StationEui missing in request");
             }
+            var stationEui = StationEui.Parse((string)rawStationEui);
```
Comment (atifaziz) [inline LoRaEngine/LoraKeysManagerFacade/ConcentratorCredentialsFunction.cs:73]: How about just going with `StationEui.TryParse` here?

```suggestion
            if (!StationEui.TryParse((string)req.Query["StationEui"], out var stationEui))
            {
                this.logger.LogError("StationEui missing in request or invalid");
                return new BadRequestObjectResult("StationEui missing in request or invalid");
            }
```

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDevice.cs [hunk 5]
```diff
@@ -48,7 +48,7 @@ public class LoRaDevice : IDisposable, ILoRaDeviceRequestQueue
 
         public AppKey? AppKey { get; set; }
 
-        public string AppEUI { get; set; }
+        public JoinEui? AppEUI { get; set; }
```
Comment (atifaziz) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDevice.cs:51]: Should we use this as an opportunity to rename `AppEUI` to the more conventional `AppEui`?

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadJoinRequest.cs [hunk 6]
```diff
@@ -15,7 +15,7 @@ public class LoRaPayloadJoinRequest : LoRaPayload
         /// <summary>
         /// Gets or sets aka JoinEUI.
         /// </summary>
-        public Memory<byte> AppEUI { get; set; }
+        public JoinEui AppEUI { get; set; }
```
Comment (atifaziz) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadJoinRequest.cs:18]: Should we use this as an opportunity to rename `AppEUI` to the more conventional `AppEui`?

File: Tests/Unit/NetworkServer/LoRaDeviceTest.cs [hunk 7]
```diff
@@ -107,11 +107,12 @@ public async Task After_Saving_Frame_Counter_Changes_Should_Not_Have_Pending_Cha
         public async Task When_Initialized_New_OTAA_Device_Should_Have_All_Properties()
         {
             var appKey = TestKeys.CreateAppKey(0xABC0200000000000, 0x09);
+            var joinEui = new JoinEui(0xABC0200000000009);
 
             var twin = TestUtils.CreateTwin(
                 desired: new Dictionary<string, object>
                 {
-                    { "AppEUI", "ABC0200000000009" },
+                    { "AppEUI", joinEui.ToString() },
```
Comment (atifaziz) [inline Tests/Unit/NetworkServer/LoRaDeviceTest.cs:115]: Shouldn't this be using the `N` format to mean the same thing?

**Review by atifaziz — APPROVED** [review]

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDevice.cs (hunk 5 above)
Comment (bastbu) [reply LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDevice.cs]: Done in 6339ab6c5009e7292142037060b8e8c60128f308

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadJoinRequest.cs (hunk 6 above)
Comment (bastbu) [reply LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadJoinRequest.cs]: Done in 6339ab6c5009e7292142037060b8e8c60128f308

File: Tests/Unit/NetworkServer/LoRaDeviceTest.cs (hunk 7 above)
Comment (bastbu) [reply Tests/Unit/NetworkServer/LoRaDeviceTest.cs]: Valid point, done in fd8bc7daed7f63f42291c59ed33ae9bcb17917e3

File: Tests/Simulation/IntegrationTestFixtureSim.cs (hunk 2 above)
Comment (bastbu) [reply Tests/Simulation/IntegrationTestFixtureSim.cs]: Resolving this as per other comment

File: Tests/Simulation/IntegrationTestFixtureSim.cs (hunk 1 above)
Comment (bastbu) [reply Tests/Simulation/IntegrationTestFixtureSim.cs]: Update: This comment caught a valid bug. I thought that devices were recreated on E2E test runs if the properties do not match (for which we have logic, which is apparently not executed on each E2E run). I needed to revert to `JoinEui.Parse("...0027")` for the E2E tests to make sure that the EUIs still match the values of the twin.

**Pull request merged**